tests/python/unittest/test_optimizer.py

```diff
@@ -520,10 +520,10 @@ def test_adam():
                                     not kwarg['multi_precision'])):
                             continue
                         compare_optimizer(opt1(**kwarg), opt2(**kwarg), shape, dtype)
-                        if (default_context() == mx.cpu()):
-                            compare_optimizer(opt1(sparse_update=True, **kwarg), opt2(**kwarg), shape,
```

Comment:
Any particular reasons to only test sparse adam on cpu?

Reply:
Nope. Maybe at that time cast_storage on gpu is not ready